daffodil-codegen-c/src/main/scala/org/apache/daffodil/codegen/c/DaffodilCCodeGenerator.scala

```diff
@@ -306,6 +310,17 @@ object DaffodilCCodeGenerator
   private def elementCombinator(g: ElementCombinator, cgState: CodeGeneratorState): Unit = {
     cgState.pushElement(g.context)
     DaffodilCCodeGenerator.generateCode(g.subComb, cgState)
+    // Also generate code for the element's assert statements
+    g.context.assertStatements.foreach { assert =>
+      assert.testKind match {
+        case TestKind.Pattern =>
```

Comment:
Is the convention for the C code generator to silently ignore things it doesn't support? I imagine an SDE or Warning might be useful in some cases, but if this is what you've been doing this is fine.